rmf_site_editor/src/widgets/console.rs

```diff
@@ -0,0 +1,471 @@
+            LogCategory::Hint => Color32::LIGHT_GREEN,
+            LogCategory::Status => Color32::WHITE,
+            LogCategory::Warning => Color32::YELLOW,
+            LogCategory::Error => Color32::RED,
+            LogCategory::Bevy => Color32::LIGHT_BLUE,
+        };
+        ui.label(RichText::new(log.category.to_string()).color(category_text_color));
+        // Selecting the label allows users to copy log entry to clipboard
+        if ui
+            .selectable_label(false, log.message.to_string())
+            .clicked()
+        {
+            ui.output().copied_text = log.category.to_string() + &log.message;
+        }
+    });
+}
+
+pub struct ConsoleWidget<'a, 'w2, 's2> {
+    events: &'a mut AppEvents<'w2, 's2>,
+}
+
+impl<'a, 'w2, 's2> ConsoleWidget<'a, 'w2, 's2> {
+    pub fn new(events: &'a mut AppEvents<'w2, 's2>) -> Self {
+        Self { events }
+    }
+
+    pub fn show(self, ui: &mut Ui) {
+        // Check updated logs
+        self.events.display.log_history.receive_log();
+
+        ui.horizontal_wrapped(|ui| {
+            ui.spacing_mut().item_spacing.x = 0.5;
+            let status = self.events.display.log_history.current_log();
+            match status {
+                Some(log) => print_log(ui, log),
+                None => (),
+            }
+        });
+        ui.add_space(5.0);
+        CollapsingHeader::new("RMF Site Editor Console")
+            .default_open(false)
+            .show(ui, |ui| {
+                ui.horizontal_wrapped(|ui| {
+                    ui.spacing_mut().item_spacing.x = 10.0;
+                    // Filter logs by category
+                    ui.checkbox(self.events.display.log_history.checked_all_mut(), "All");
+                    ui.checkbox(
+                        self.events
+                            .display
+                            .log_history
+                            .category_present_mut(LogCategory::Status),
+                        "Status",
+                    );
+                    ui.checkbox(
+                        self.events
+                            .display
+                            .log_history
+                            .category_present_mut(LogCategory::Warning),
+                        "Warning",
+                    );
+                    ui.checkbox(
+                        self.events
+                            .display
+                            .log_history
+                            .category_present_mut(LogCategory::Error),
+                        "Error",
+                    );
+                    ui.checkbox(
+                        self.events
+                            .display
+                            .log_history
+                            .category_present_mut(LogCategory::Bevy),
+                        "Bevy",
+                    );
+                    // Copy full log history to clipboard
+                    if ui.button("Copy Log History").clicked() {
```

Comment:
Not sure this works, I tried and all I could get is exporting of only the first message in the log

Reply:
Ah, i think this only works on locally built editor. I could copy a single/full log and paste it into any open-ended field within the web app (example the `name` field of the Level), but not outside of the browser. Will look into this further

Reply:
Even in the local built editor I can only get it to copy the first message, I tried to click on a few messages as well but it doesn't seem to work

Reply:
Oh that's really weird, it seems to work for me. Also works on VM too. Let me try with another machine

Reply:
Could you try a `rustup update`? It wasn't working on another machine and after an update it works now

Reply:
We looked into it a bit more and it seems to be a clipboard issue and depends on the software that is pasted on, a few (non exhaustive) examples:

* Chrome works, but firefox doesn't
* VS code works, but vim, thunar and gedit don't

I suspect this might be a bit too hairy to solve

Reply:
As Luca pointed out there was a `\0` hidden in the tracing layer. Removed in [3ccac6d](https://github.com/open-rmf/rmf_site/pull/121/commits/3ccac6d0bb69be22becda93643c8e151626d23c3) and it should copy properly now.